In [21]:
from scipy.stats import norm 
from scipy import stats 
import pandas as pd 
import numpy as np

# Q 1] 

In [2]:
mu = 45
sigma = 8
x = 50
print(f'Probability of service manager not meeting his commitment {round(1-norm.cdf(x,loc=mu,scale=sigma),3)}')

Probability of service manager not meeting his commitment 0.266


### Probability of service manager not meeting his commitment 0.266

# Q 2] A]

In [3]:
mu = 38
sigma = 6
# probability that employees are older than 44 
A = 1-norm.cdf(44,loc=mu,scale=sigma)
# probability that age of employees is between 38 and 44
B = norm.cdf(44,loc=mu,scale=sigma)-norm.cdf(38,loc=mu,scale=sigma)
pd.DataFrame([['P(age>44)',A],
              ['P(38<age<44)',B]],columns=['condition','Probabiliy'])

condition  Probabiliy
0     P(age>44)    0.158655
1  P(38<age<44)    0.341345

# Q 2] B]

In [4]:
n = 400
mu = 38
sigma = 6
# probability that age of employee is less than 30
prob = norm.cdf(30,loc=mu,scale=sigma)
# expected number of employees 
exp = round(prob*n)
print(f'training program for employees under the age of 30 at the center would be expected to attract about {exp} employees.')

training program for employees under the age of 30 at the center would be expected to attract about 36 employees.


# Q 4]

In [8]:
confLevel = 0.99
mu = 100
sigma = 20
# confidence interval for confidence level of 99 %
r = norm.interval(confidence=confLevel,
                  loc=mu,
                  scale=sigma)
print(f'for [X ~ N(100, 202)] range where probability of random variable taking value is 0.99 is [{round(r[0],1)}, {round(r[1],1)}]')

for [X ~ N(100, 202)] range where probability of random variable taking value is 0.99 is [48.5, 151.5]


# Q 5] A]

In [25]:
mu1 = 5
var1 = 3
mu2 = 7
var2 = 4
# variance of combined normal distributions of 
# profits from two 'independent' divisions 
combVar = var1 + var2 
combSigma = combVar**0.5
combMu = mu1 + mu2 
confiLevel = 0.95
# confidence interval for confidence level of 95 %
r = norm.interval(confidence=confiLevel,
                  loc=combMu,
                  scale=combSigma)
print(f'Rupee range is from {round(r[0]*45,2)} million rupees to {round(r[1]*45,2)} million rupees')

Rupee range is from 306.65 million rupees to 773.35 million rupees


# Q 5] B]

In [35]:
# calculating 5th percentile 
fifthP = round(norm.ppf(0.05,loc=combMu,scale=combSigma)*45,2)
print(f'5th percentile profit for the company is {fifthP} million rupees')

5th percentile profit for the company is 344.17 million rupees


# Q 5] C]

In [20]:
# probability that division 1 will make loss
p1_loss = norm.cdf(0,loc=mu1,scale=var1**0.5)
print(f'probability that division 1 will make loss is {p1_loss}')
# probability that division 2 will make loss
p2_loss = norm.cdf(0,loc=mu2,scale=var2**0.5)
print(f'probability that division 2 will make loss is {p2_loss}')

probability that division 1 will make loss is 0.001946208561389314
probability that division 2 will make loss is 0.00023262907903552502
